### PR TITLE
Add database stuff

### PR DESCRIPTION
**Pull request by NYC1809**

### CODE REVIEW

File: src/main/kotlin/de/skycave/shoprotation/model/ChestItems.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    lateinit var name: String
+    lateinit var items: Map<Material, Int>
+
+    constructor()
+
+    constructor(id: ObjectId, name: String, items: Map<Material, Int>) {
+        this.id = id
+        this.name = name
+        this.items = items
+
```
Comment (heuerleon) [inline src/main/kotlin/de/skycave/shoprotation/model/ChestItems.kt:22]: Here is a blank line too much :D

File: src/main/kotlin/de/skycave/shoprotation/codecs/SingleIntCodec.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class SingleIntCodec: Codec<Single<Int>> {
+    override fun encode(writer: BsonWriter?, value: Single<Int>?, encoderContext: EncoderContext?) {
+        writer ?: return
+        if (value == null) {
+            writer.writeNull()
+            return
+        }
+
+        writer.writeInt32(value.value)
+
```
Comment (heuerleon) [inline src/main/kotlin/de/skycave/shoprotation/codecs/SingleIntCodec.kt:23]: Here too

**Review by heuerleon — CHANGES_REQUESTED** [review]

File: src/main/kotlin/de/skycave/shoprotation/model/ChestItems.kt (hunk 1 above)
Comment (NYC1809) [reply src/main/kotlin/de/skycave/shoprotation/model/ChestItems.kt:22]: no

File: src/main/kotlin/de/skycave/shoprotation/codecs/SingleIntCodec.kt (hunk 2 above)
Comment (NYC1809) [reply src/main/kotlin/de/skycave/shoprotation/codecs/SingleIntCodec.kt:23]: no

File: src/main/kotlin/de/skycave/shoprotation/model/ChestItems.kt (hunk 1 above)
Comment (NYC1809) [reply src/main/kotlin/de/skycave/shoprotation/model/ChestItems.kt]: fixed

File: src/main/kotlin/de/skycave/shoprotation/codecs/SingleIntCodec.kt (hunk 2 above)
Comment (NYC1809) [reply src/main/kotlin/de/skycave/shoprotation/codecs/SingleIntCodec.kt]: fixed

**Review by heuerleon — APPROVED** [review]
Looks good

**Pull request merged**